apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerConnection.cs

```diff
@@ -639,6 +639,18 @@ private Task OpenAsyncImpl(Action transactionEnlister, CancellationToken cancell
                     OnStateChange(new StateChangeEventArgs(previousState, ConnectionState.Connecting));
                     try
                     {
+                        bool isResourceBasedRoutingEnabled = GetResourceBasedRoutingFlag();
```

Comment:
All of this logic should be moved to the `InstanceHostManage` class as well.

Reply:
Done!!